### DISCUSSION TITLE
Access deleted entity

### DISCUSSION THREAD

**Discussion by swilly22** [FalkorDB/FalkorDB#16] [Polls]
The rules for accessing a deleted entity aren't well defined.

Consider: `MATCH (a) DELETE a RETURN a`
Should `a` be returned? if so are we expecting A's label(s), ID and attributes to return?

What about this query: `MATCH (a:A), (b:B) DELETE a SET b.v = a.v`
Do we expect `b.v` to take `a.v` original value ?

As I see it there are two options:
1. Any access to a deleted entity will cause the query to fail.
2. Once an entity is deleted its attribute-set becomes immutable, reading data from it is allowed.

Thoughts?

**Comment by DvirDukhan** [comment]
@swilly22 please check open cypher deleted entities persistency tests https://github.com/opencypher/openCypher/blob/master/tck/features/clauses/delete/Delete6.feature

edit:
also take a look at the remove clause https://github.com/opencypher/openCypher/blob/master/tck/features/clauses/remove/Remove1.feature
I would suggest keeping the same (implied) semantics, meaning, the entity does not exists post delete clause

**Reply by swilly22** [reply]
Unfortunately the TCK doesn't touch on the "interesting" cases where the deleted entity is accessed after its deletion.

**Reply by swilly22** [reply]
Indeed the `REMOVE` attribute semantics are clear.
